airflow/providers/imap/hooks/imap.py

```diff
@@ -71,7 +71,7 @@ def get_conn(self) -> 'ImapHook':
         """
         if not self.mail_client:
             conn = self.get_connection(self.imap_conn_id)
-            self.mail_client = imaplib.IMAP4_SSL(conn.host, conn.port or imaplib.IMAP4_SSL_PORT)
+            self.mail_client = imaplib.IMAP4_SSL(conn.host, conn.port or imaplib.IMAP4_SSL.port)
```

Comment:
I think this solution might be not correct.
There is no class attribute for `IMAP4_SSL`, and `self.port` is defined in method `open`.
See how I decided to resolve this issue
https://github.com/apache/airflow/pull/20654/files#diff-b986a5d7078309e7eaad82d9b534ead3b21f0f884c927fdd21674cd4390a935a

Reply:
I was going off this, the default value for port in the constructor for IMAP4_SSL 
```

    class IMAP4_SSL(IMAP4):

        """IMAP4 client class over SSL connection

        Instantiate with: IMAP4_SSL([host[, port[, keyfile[, certfile[, ssl_context]]]]])

                host - host's name (default: localhost);
                port - port number (default: standard IMAP4 SSL port);
                keyfile - PEM formatted file that contains your private key (default: None);
                certfile - PEM formatted certificate chain file (default: None);
                ssl_context - a SSLContext object that contains your certificate chain
                              and private key (default: None)
                Note: if ssl_context is provided, then parameters keyfile or
                certfile should not be set otherwise ValueError is raised.

        for more documentation see the docstring of the parent class IMAP4.
        """


        def __init__(self, host='', port=IMAP4_SSL_PORT, keyfile=None,
                     certfile=None, ssl_context=None):
```

Reply:
I can close this PR, this was supposed to be minor changes.

Reply:
Yeap, if we do not pass the port in the constructor then the default will be used.
I even wanted to suggest `__all__.extend("IMAP4_SSL", "IMAP4_SSL_PORT"])` here 
https://github.com/python/cpython/blob/7d7817cf0f826e566d8370a0e974bbfed6611d91/Lib/imaplib.py#L1338
but decided that these changes won't be applied soon

Let's just wait for the review and will see

Reply:
I don’t think `IMAP4_SSL.port` is going to work. Whether `IMAP4_SSL_PORT` is public or not is slightly ambiguous; it is not by module definition, [but the variable actually appears in documentation](https://docs.python.org/3/library/imaplib.html#imaplib.IMAP4_SSL). Either case, the IMAP-over-SSL port is standardised (993), so maybe we should just not rely on `imaplib.IMAP4_SSL_PORT` and just define our own variable.

Reply:
The `IMAP4_SSL_PORT` is not public, but it is a default port when the only the host is passing to the constructor

Reply:
My solution is
```
conn = self.get_connection(self.imap_conn_id)
if conn.port:
    self.mail_client = imaplib.IMAP4_SSL(conn.host, conn.port)
else:
    self.mail_client = imaplib.IMAP4_SSL(conn.host)
self.mail_client.login(conn.login, conn.password)
```

Reply:
Looks like a good alternative solution to me!

Reply:
Thanks, updated.